### PR TITLE
Droped Psalm

### PR DESCRIPTION
**Pull request by driehle**
Psalm is dropped in favour of PHPStan, according to Doctrine's upstream workflows.

### CODE REVIEW

File: .github/workflows/psalm.yml [hunk 1]
```diff
@@ -1,30 +1,28 @@
...
       - "phpstan*"
-      - "psalm*"
       - "tests/**"
 
 jobs:
-  static-analysis:
+  phpstan:
     name: "Static Analysis"
-    uses: "doctrine/.github/.github/workflows/static-analysis.yml@6.0.0"
+    uses: "doctrine/.github/.github/workflows/psalm.yml@6.0.0"
```
Comment (greg0ire) [inline .github/workflows/psalm.yml]:
```suggestion
    uses: "doctrine/.github/.github/workflows/phpstan.yml@6.0.0"
```

File: .github/workflows/psalm.yml [hunk 2]
```diff
@@ -1,30 +1,28 @@
...
       - "src/**"
       - "phpstan*"
-      - "psalm*"
       - "tests/**"
   push:
     branches:
       - "*.x"
     paths:
-      - ".github/workflows/static-analysis.yml"
+      - ".github/workflows/psalm.yml"
```
Comment (greg0ire) [inline .github/workflows/psalm.yml]:
```suggestion
      - ".github/workflows/phpstan.yml"
```

File: .github/workflows/psalm.yml [hunk 3]
```diff
@@ -1,30 +1,28 @@
-name: "Static Analysis"
+name: "Static Analysis with PHPStan"
 
 on:
   pull_request:
     branches:
       - "*.x"
     paths:
-      - ".github/workflows/static-analysis.yml"
+      - ".github/workflows/psalm.yml"
```
Comment (greg0ire) [inline .github/workflows/psalm.yml]:
```suggestion
      - ".github/workflows/phpstan.yml"
```

File: .github/workflows/psalm.yml (hunk 1 above)
Comment (driehle) [reply .github/workflows/psalm.yml]: @greg0ire Thanks, btw there is a typo in line 27 of the template in https://github.com/doctrine/.github/blob/main/workflow-templates/phpstan.yml.

**Comment by greg0ire** [comment]
There are occurrences of `psalm` in hidden files such as `.gitattributes` 

Here is a list of steps you might want to know about: https://github.com/doctrine/.github/releases/tag/5.2.0

**Comment by greg0ire** [comment]
There are more occurrences (annotations). Please use search and replace :stuck_out_tongue:

**Review by greg0ire — APPROVED** [review]

**Pull request merged**